### PR TITLE
Experimental AMD extensions

### PR DESCRIPTION
**Pull request by msiglreith**
Addresses https://github.com/MaikKlein/ash/issues/178#issuecomment-457942089

Not fully supports all parts of the APIs, just trying to get some general feedback if this meets the expected structuring.

### CODE REVIEW

File: ash/src/extensions/experimental/amd.rs [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+pub unsafe trait ExtendsPhysicalDeviceWaveLimitPropertiesAmd {}
+unsafe impl ExtendsPhysicalDeviceProperties2 for PhysicalDeviceWaveLimitPropertiesAmd {}
+impl<'a> ::std::ops::Deref for PhysicalDeviceWaveLimitPropertiesAmdBuilder<'a> {
+    type Target = PhysicalDeviceWaveLimitPropertiesAmd;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+impl<'a> PhysicalDeviceWaveLimitPropertiesAmdBuilder<'a> {
+    pub fn next<T>(mut self, next: &'a mut T) -> PhysicalDeviceWaveLimitPropertiesAmdBuilder<'a>
```
[the remark this answers is not in the document]
Comment (MaikKlein) [reply ash/src/extensions/experimental/amd.rs]: Just leave it as it is right now.

**Comment by msiglreith** [comment]
Should be ready now. I skipped the high level API as I'm not sure if it would be of much worth for now

**Comment by msiglreith** [comment]
A bit, mostly trying to make sense out of the exposed counters as 90% of these are not documented and GCN generation specific 😅 
Querying the device capabilities works fine so far at least. I need to look deeper inside GPA.

**Pull request merged**